exporter/xray/xray.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package xray contains an trace exporter for AWS X-Ray.
+package xray
+
+import (
+	"io"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/aws/aws-sdk-go/service/xray/xrayiface"
+	"go.opencensus.io/trace"
+)
+
+type OnExport struct {
+	TraceID string
+}
+
+// Options contains a set of optional configuration parameters
+type Options struct {
+	// ServiceName - optional name of the service
+	ServiceName string
+
+	// ServiceVersion - optional version of service
+	ServiceVersion string
+
+	// Output - optional writer for error messages
+	Output io.Writer
+
+	// XRayAPI - optionally allows caller to provide their own xray implementation
+	XRayAPI xrayiface.XRayAPI
+
+	// OnExport optional function to be called when spans are published to AWS; useful
+	// if you would like the traceID used for AWS
+	OnExport func(OnExport)
+}
+
+// Exporter is an implementation of trace.Exporter that uploads spans to a
+// Zipkin server.
+type Exporter struct {
+	api      xrayiface.XRayAPI
+	onExport func(export OnExport)
+	logger   *log.Logger
+	service  *service
+
+	mutex  sync.Mutex
+	traces map[trace.TraceID][]*trace.SpanData
+}
+
+// NewExporter returns an implementation of trace.Exporter that uploads spans
+// to AWS X-Ray
+func NewExporter(options Options) (*Exporter, error) {
+	if options.Output == nil {
+		options.Output = os.Stderr
+	}
+	if options.OnExport == nil {
+		options.OnExport = func(export OnExport) {}
+	}
+	if options.XRayAPI == nil {
+		s, err := session.NewSession(nil)
```

Comment:
seems like this is a variadic method, why not just pass no arguments?

Reply:
right-o!